### PR TITLE
Use new display_gross_prices flag

### PR DESCRIPTION
**Pull request by maarcingebala**
Deprecate `display_gross_prices` from site settings in favor of `display_gross_prices` from tax configuration, either global, or country-specific. Refactor all usages of the old setting to use the new approach.

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-6794 Add logic to use flat tax rates in Saleor calculations](https://app.clickup.com/t/2549495/SALEOR-6794)

File: saleor/graphql/checkout/tests/mutations/test_checkout_add_promo_code.py [hunk 1]
```diff
@@ -197,8 +197,11 @@ def test_checkout_add_voucher_code_by_token_with_external_shipment(
 def test_checkout_add_voucher_code_with_display_gross_prices(
     api_client, checkout_with_item, voucher, site_settings, monkeypatch
 ):
-    site_settings.display_gross_prices = True
-    site_settings.save()
+    channel = checkout_with_item.channel
+    tc = channel.tax_configuration
+    tc.display_gross_prices = True
+    tc.save()
```
Comment (IKarbowiak) [inline saleor/graphql/checkout/tests/mutations/test_checkout_add_promo_code.py:203]:
```suggestion
    tc.save(update_fields=["display_gross_prices"])
```

File: saleor/graphql/checkout/tests/mutations/test_checkout_add_promo_code.py [hunk 2]
```diff
@@ -230,8 +233,11 @@ def test_checkout_add_voucher_code_with_display_gross_prices(
 def test_checkout_add_voucher_code_without_display_gross_prices(
     api_client, checkout_with_item, voucher, site_settings, monkeypatch
 ):
-    site_settings.display_gross_prices = False
-    site_settings.save()
+    channel = checkout_with_item.channel
+    tc = channel.tax_configuration
+    tc.display_gross_prices = False
+    tc.save()
```
Comment (IKarbowiak) [inline saleor/graphql/checkout/tests/mutations/test_checkout_add_promo_code.py:239]: Same here 👮‍♀️

File: saleor/order/tests/test_order_utils.py [hunk 3]
```diff
@@ -588,3 +589,42 @@ def test_get_total_order_discount_excluding_shipping_no_shipping_discounts(
...
+
+def test_update_order_display_gross_prices_use_default_tax_settings(order):
+    # given
+    tax_config = order.channel.tax_configuration
+    tax_config.display_gross_prices = True
+    tax_config.save()
+    tax_config.country_exceptions.all().delete()
+
+    order.display_gross_prices = False
+    order.save()
```
Comment (IKarbowiak) [inline saleor/order/tests/test_order_utils.py:602]: Similar with order
```suggestion
    order.save(update_fields=["display_gross_prices"])
```

**Review by IKarbowiak — APPROVED** [review]

File: saleor/graphql/checkout/tests/mutations/test_checkout_add_promo_code.py (hunk 1 above)
Comment (jakubkuc) [reply saleor/graphql/checkout/tests/mutations/test_checkout_add_promo_code.py:203]: in `test_discounts.py` line 32 and 54 as well

**Review by jakubkuc — APPROVED** [review]

File: saleor/graphql/order/mutations/order_line_update.py [hunk 4]
```diff
@@ -81,8 +81,8 @@ def save(cls, info, instance, cleaned_input):
                 "Cannot set new quantity because of insufficient stock.",
                 code=OrderErrorCode.INSUFFICIENT_STOCK,
             )
-        recalculate_order(instance.order)
 
+        recalculate_order(instance.order)
```
Comment (fowczarek) [inline saleor/graphql/order/mutations/order_line_update.py:85]: We could leave this file unchanged.

File: saleor/graphql/shop/types.py [hunk 5]
```diff
@@ -215,7 +220,12 @@ class Shop(graphene.ObjectType):
         required=True,
     )
     display_gross_prices = graphene.Boolean(
-        description="Display prices with tax in store.", required=True
+        description="Display prices with tax in store.",
+        deprecation_reason=(
+            f"{DEPRECATED_IN_3X_FIELD} Use `Channel.taxConfiguration` to determine "
```
Comment (fowczarek) [inline saleor/graphql/shop/types.py:225]: Maybe? 
```suggestion
            f"{DEPRECATED_IN_3X_FIELD} Use `Channel.taxConfiguration.displayGrossPrices` to determine "
```

File: saleor/order/models.py [hunk 6]
```diff
@@ -278,7 +278,6 @@ class Order(ModelWithMetadata):
         Voucher, blank=True, null=True, related_name="+", on_delete=models.SET_NULL
     )
     gift_cards = models.ManyToManyField(GiftCard, blank=True, related_name="orders")
-
```
Comment (fowczarek) [inline saleor/order/models.py:281]: We could leave this file unchanged.

**Review by fowczarek — APPROVED** [review]

File: saleor/graphql/shop/types.py (hunk 5 above)
Comment (maarcingebala) [reply saleor/graphql/shop/types.py:225]: `Channel.taxConfiguration.displayGrossPrices` is not the only place where you configure it; there is also a country-specific configuration (returned as a list of items) available in this object. So I think you need to check this entire object to configure it as you wish.

**Pull request merged**